Subject: [PATCH] Fix megaraid oops on unload
X-BK-ChangeSetKey: viro@parcelfarce.linux.theplanet.co.uk[torvalds]|ChangeSet|20050105191130|64346
Date:	Wed, 05 Jan 2005 21:03:00 +0000
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.2217, 2005/01/05 11:11:30-08:00, viro@parcelfarce.linux.theplanet.co.uk

	[PATCH] Fix megaraid oops on unload
	
	Reported by Wakko Warner.
	
	The megaraid driver is trying to remove a non-empty directory in procfs.
	
	Trivially fixed by first unregistering the driver, which will remove all
	the per-controller files in the megaraid directory, and only _then_
	removing the megaraid /proc directory entry.



 megaraid.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)


diff -Nru a/drivers/scsi/megaraid.c b/drivers/scsi/megaraid.c
--- a/drivers/scsi/megaraid.c	2005-01-05 12:17:43 -08:00
+++ b/drivers/scsi/megaraid.c	2005-01-05 12:17:43 -08:00
@@ -5109,11 +5109,11 @@
 	 */
 	unregister_chrdev(major, "megadev");
 
+	pci_unregister_driver(&megaraid_pci_driver);
+
 #ifdef CONFIG_PROC_FS
 	remove_proc_entry("megaraid", &proc_root);
 #endif
-
-	pci_unregister_driver(&megaraid_pci_driver);
 }
 
 module_init(megaraid_init);
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

